### PR TITLE
pass firefox browsercontext.spec.js tests

### LINKED ISSUE #92: Pass browsercontext.spec.js tests in Firefox

### PR DESCRIPTION
**Pull request by Meir017**
closes #92

### CODE REVIEW

File: src/PlaywrightSharp.Firefox/FirefoxTarget.cs [hunk 1]
```diff
@@ -60,39 +56,49 @@ public Task<IWorker> GetWorkerAsync()
 
         internal void DidClose() => _page?.DidClose();
 
-        private async Task<Page> CreatePageAsync()
+        internal Task<Page> CreatePageAsync()
```
Comment (Meir017) [inline src/PlaywrightSharp.Firefox/FirefoxTarget.cs]: @kblok  kind of stuck with this

Comment (kblok) [reply src/PlaywrightSharp.Firefox/FirefoxTarget.cs]: I'll take a look 👍

**Comment by kblok** [comment]
@Meir017 I didn't know I was able to push to your branch 🤕 sorry about that. But well. some progress in here.

**Comment by Meir017** [comment]
@kblok thanks for the help on this, it's ready for review :)

File: src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs [hunk 2]
```diff
@@ -10,51 +10,89 @@ namespace PlaywrightSharp.Firefox
...
+
+            return (includeType ? "JSHandle:" : string.Empty) + DeserializeValue<object>((RemoteObject)payload);
         }
 
         public Task<T> HandleJSONValueAsync<T>(IJSHandle jsHandle) => throw new NotImplementedException();
 
         public Task ReleaseHandleAsync(JSHandle handle)
         {
-            throw new System.NotImplementedException();
+            if (handle?.RemoteObject?.ObjectId == null)
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/FirefoxExecutionContext.cs:71]:
```suggestion
            if (string.IsNullOrEmpty(handle?.RemoteObject.ObjectId))
```

File: src/PlaywrightSharp.Firefox/Protocol/BoundingBox.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+namespace PlaywrightSharp.Firefox.Protocol.Page
+{
+    internal partial class BoundingBox
+    {
+        public static implicit operator BoundingBox(Rect rect) => rect == null ? null : new BoundingBox
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/Protocol/BoundingBox.cs:5]: arrow in next line?

File: src/PlaywrightSharp.Firefox/Protocol/BoundingBox.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            X = rect.X,
+            Y = rect.Y,
+            Height = rect.Height,
+            Width = rect.Width,
+        };
+
+        public static implicit operator Rect(BoundingBox boundingBox) => boundingBox == null ? null : new Rect
```
Comment (kblok) [inline src/PlaywrightSharp.Firefox/Protocol/BoundingBox.cs:13]: arrow in next line?

**Pull request merged**